doc/release/upcoming_changes/24587.python_removal.rst

```diff
@@ -0,0 +1,8 @@
+* ``np.compare_chararrays`` has been removed from the main namespace. 
+  Use ``np.char.compare_chararrays`` instead.
+
+* ``np.chararray`` has been deprecated and it will be removed in the future.
+  Use ``np.char.chararray`` instead.
```

Comment:
I think I would rephrase this slightly as "The `charrarray` in the main namespace has been deprecated. It can be imported without a deprecation warning from `np.char.chararray` for now, but we are planning to fully deprecate and remove `chararray` in the future.", since I think the intention is to fully deprecate it and remove it, we just haven't actually done the work yet, so can't outright remove it now. I proposed doing those deprecations before NumPy 2.0 as part of NEP 55, but that hasn't fully gone through yet. If that happens before NumPy 2.0, I will update this release note.